### PR TITLE
[SYCLomatic] Fix the migration of curandCreateGeneratorHost

### PR DESCRIPTION
**Pull request by zhiweij1**
Signed-off-by: Jiang, Zhiwei <zhiwei.jiang@intel.com>

### CODE REVIEW

File: clang/runtime/dpct-rt/include/dpct/rng_utils.hpp [hunk 1]
```diff
@@ -502,30 +505,31 @@ typedef std::shared_ptr<rng::host::detail::rng_generator_base> host_rng_ptr;
 /// Create a host random number generator.
```
Comment (tangjj11) [inline clang/runtime/dpct-rt/include/dpct/rng_utils.hpp:509]: Need add comment `/// \tparam work_on_cpu ...`

**Review by tangjj11 — APPROVED** [review]

File: clang/runtime/dpct-rt/include/dpct/rng_utils.hpp (hunk 1 above)
Comment (zhiweij1) [reply clang/runtime/dpct-rt/include/dpct/rng_utils.hpp:509]: Updated.

File: clang/test/dpct/curand.cu [hunk 2]
```diff
@@ -251,6 +251,6 @@ void bar7() {
   curandRngType_t rngT9 = CURAND_RNG_QUASI_SCRAMBLED_SOBOL32;
   curandRngType_t rngT10 = CURAND_RNG_QUASI_SOBOL64;
   curandRngType_t rngT11 = CURAND_RNG_QUASI_SCRAMBLED_SOBOL64;
-  // CHECK: rng = dpct::rng::create_host_rng(rngT);
+  // CHECK: rng = dpct::rng::create_host_rng<true>(rngT);
```
Comment (zhiweij1) [inline clang/test/dpct/curand.cu]: Add a default parameter default_queue

File: clang/runtime/dpct-rt/include/dpct/rng_utils.hpp [hunk 3]
```diff
@@ -209,6 +209,10 @@ namespace host {
 namespace detail {
 class rng_generator_base {
 public:
+  /// Construct the host rng_generator.
```
Comment (zhiweij1) [inline clang/runtime/dpct-rt/include/dpct/rng_utils.hpp]: move to protected

File: clang/runtime/dpct-rt/include/dpct/rng_utils.hpp [hunk 4]
```diff
@@ -500,32 +506,36 @@ enum class random_engine_type {
...
 
 /// Create a host random number generator.
+/// \tparam work_on_cpu Whether the work is offloaded to CPU.
 /// \param type The random engine type.
+/// \param q The queue where the generator should be executed.
 /// \return The pointer of random number generator.
-inline host_rng_ptr create_host_rng(const random_engine_type type) {
+inline host_rng_ptr
+create_host_rng(const random_engine_type type,
+                sycl::queue *q = &dpct::get_default_queue()) {
```
Comment (zhiweij1) [inline clang/runtime/dpct-rt/include/dpct/rng_utils.hpp]:
```suggestion
                sycl::queue& q = dpct::get_default_queue()) {
```

**Review by ziranzha — APPROVED** [review]

**Pull request merged**